src/middlewared/middlewared/plugins/disk_/disk_info.py

```diff
@@ -7,6 +8,31 @@
 from middlewared.service import CallError, private, Service
 
 
+def get_partition_size_info(disk_name, s_offset, s_size):
+    """Kernel sysfs reports most disk files related to "size" in 512 bytes.
+    To properly calculate the starting SECTOR of partitions, you must
+    look at logical_block_size (again, reported in 512 bytes) and
+    do some calculations. It is _very_ important to do this properly
+    since almost all userspace tools that format disks expect partition
+    positions to be in sectors."""
+    lbs = start_sect = end_sect = sect_size = 0
+    with contextlib.suppress(FileNotFoundError, ValueError):
+        with open(f'/sys/block/{disk_name}/queue/logical_block_size') as f:
+            lbs = int(f.read().strip())
+
+    if not lbs:
+        # this should never happen
+        return lbs, start_sect, end_sect, sect_size
+
+    # important when dealing with 4kn drives
+    divisor = lbs // 512
+    start_sect = s_offset // divisor
+    sect_size = s_size // divisor
+    end_sect = sect_size + start_sect - 1
+
+    return lbs, start_sect, end_sect, sect_size
```

Comment:
It would be great to return a namedtuple instead

Reply:
good idea, I'll add that